src/compiler/parser.ts

```diff
@@ -3480,7 +3518,7 @@ module ts {
                 case SyntaxKind.VoidKeyword:
                     return parseVoidExpression();
                 case SyntaxKind.LessThanToken:
-                    return parseTypeAssertion();
+                    return (sourceFile.jsxFactory && tryParse(() => parseJSXElement(true))) || parseTypeAssertion();

```

Comment:
in this case for example: 
```
<div> Some jsx text { someExpression}  ....... </dv> 
```
If we look for a closing `</div>` tag before parsing the content of the element for validating that `<div>` is not a type assertion, everything inside the element, and especially JSX text, will be parsed as normal code, which generally lead in a lot of errors, and not reported at the right place, when the true error is that there is a typo in the closing element tag name.